### PR TITLE
cleanups to proxy change trackers

### PR DESCRIPTION
**Pull request by danwinship**
#### What type of PR is this?
/kind cleanup
/sig network
/triage accepted

#### What this PR does / why we need it:
A few cleanups to ServiceChangeTracker / EndpointsChangeTracker. (prereqs to other changes coming later)

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/proxy/endpointschangetracker.go [hunk 1]
```diff
@@ -62,12 +60,18 @@ type makeEndpointFunc func(info *BaseEndpointInfo, svcPortName *ServicePortName)
 type processEndpointsMapChangeFunc func(oldEndpointsMap, newEndpointsMap EndpointsMap)
 
 // NewEndpointsChangeTracker initializes an EndpointsChangeTracker
-func NewEndpointsChangeTracker(hostname string, makeEndpointInfo makeEndpointFunc, ipFamily v1.IPFamily, recorder events.EventRecorder, processEndpointsMapChange processEndpointsMapChangeFunc) *EndpointsChangeTracker {
+func NewEndpointsChangeTracker(hostname string, makeEndpointInfo makeEndpointFunc, ipFamily v1.IPFamily, _ events.EventRecorder, processEndpointsMapChange processEndpointsMapChangeFunc) *EndpointsChangeTracker {
```
Comment (aojea) [inline pkg/proxy/endpointschangetracker.go]: why not just remove the `events.EventRecorder`?

Comment (aojea) [reply pkg/proxy/endpointschangetracker.go]: duh, next commit

Comment (danwinship) [reply pkg/proxy/endpointschangetracker.go]: yeah, I didn't want to have to rewrite all the callers twice

File: pkg/proxy/endpointslicecache.go [hunk 2]
```diff
@@ -72,16 +67,14 @@ type endpointSliceData struct {
 }
 
 // NewEndpointSliceCache initializes an EndpointSliceCache.
-func NewEndpointSliceCache(hostname string, ipFamily v1.IPFamily, recorder events.EventRecorder, makeEndpointInfo makeEndpointFunc) *EndpointSliceCache {
+func NewEndpointSliceCache(hostname string, makeEndpointInfo makeEndpointFunc) *EndpointSliceCache {
```
Comment (aojea) [inline pkg/proxy/endpointslicecache.go:70]: commented offline, now the EndpointSliceCache is family unaware, so it is caller responsability to guarantee it only receives one family

Comment (aojea) [reply pkg/proxy/endpointslicecache.go:70]: self note

**Comment by danwinship** [comment]
oops, rebased to take into account the new unit tests from #128929

**Comment by aojea** [comment]
logs show the symptoms of the kernel bug, but don't use to timeout the job

**Comment by aojea** [comment]
it started to fail more since yesterday, timing out https://prow.k8s.io/job-history/gs/kubernetes-ci-logs/pr-logs/directory/pull-kubernetes-e2e-kind-nftables but is failing in other PRs, I would be surprised this change impact only one proxy, if that is true we have a problem something is leaking

/lgtm

let's iterate

**Pull request merged**